instance.tf

```diff
@@ -3,45 +3,85 @@ resource "google_sql_database_instance" "instance" {
   name                 = var.instance_name
   deletion_protection  = var.deletion_protection
   master_instance_name = var.primary_instance_name
+
   settings {
+    activation_policy     = "ALWAYS"
     availability_type     = var.highly_available == true ? "REGIONAL" : "ZONAL"
     disk_autoresize       = var.storage_autoresize
     disk_autoresize_limit = var.storage_limit
     disk_size             = local.storage_size
     disk_type             = "PD_SSD"
+    pricing_plan          = "PER_USE"
     tier                  = local.tier
     user_labels           = local.labels
+
     backup_configuration {
-      enabled  = local.backup_config.enabled
-      location = local.backup_config.location
+      enabled                        = local.backup_config.enabled
+      start_time                     = local.backup_config.start_time
+      point_in_time_recovery_enabled = local.backup_config.point_in_time_recovery_enabled
+      location                       = local.backup_config.location
+      transaction_log_retention_days = local.backup_config.transaction_log_retention_days
+      dynamic "backup_retention_settings" {
+        for_each = local.needs_backup_retention_settings
+
+        content {
+          retained_backups = local.backup_config.retained_backups
+          retention_unit   = "COUNT"
+        }
+      }
```

Comment:
Wouldn't it make more sense to not make this block optional, but instead always set `retained_backups` to a sane (default) value?

Reply:
Perhaps, yes. I couldn't determine GCP default, which is why I made it optional. Any thoughts on a sane default?

Reply:
See other remark regarding using the state to try to figure out the current settings aka defaults.